src/rulesets/Base/arraymath.jl

```diff
@@ -19,23 +19,47 @@ end
 ##### `*`
 #####
 
-function rrule(::typeof(*), A::AbstractMatrix{<:Real}, B::AbstractMatrix{<:Real})
+function rrule(::typeof(*), A::AbstractMatrix{<:Number}, B::AbstractMatrix{<:Number})
```

Comment:
@sethaxen pointed out that we probably want to restrict these to `Real` and `Complex` numbers.